modules/trino/src/main/java/org/testcontainers/containers/TrinoContainer.java

```diff
@@ -0,0 +1,109 @@
+package org.testcontainers.containers;
+
+import org.jetbrains.annotations.NotNull;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static java.lang.String.format;
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+/**
+ * @author findepi
+ */
+public class TrinoContainer<SELF extends TrinoContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
+    public static final String NAME = "trino";
+    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("trinodb/trino");
+    public static final String IMAGE = "trinodb/trino";
+    public static final String DEFAULT_TAG = "351";
+
+    public static final Integer TRINO_PORT = 8080;
+
+    private String username = "test";
+    private String catalog = null;
+
+    public TrinoContainer(final String dockerImageName) {
+        this(DockerImageName.parse(dockerImageName));
+    }
+
+    public TrinoContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+
+        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
+
+        this.waitStrategy = new LogMessageWaitStrategy()
+            .withRegEx(".*======== SERVER STARTED ========.*")
+            .withStartupTimeout(Duration.of(60, SECONDS));
+
+        addExposedPort(TRINO_PORT);
+    }
+
+    @NotNull
+    @Override
+    protected Set<Integer> getLivenessCheckPorts() {
+        return new HashSet<>(getMappedPort(TRINO_PORT));
+    }
+
+    @Override
+    public String getDriverClassName() {
+        return "io.trino.jdbc.TrinoDriver";
+    }
+
+    @Override
+    public String getJdbcUrl() {
+        return format("jdbc:trino://%s:%s/%s", getHost(), getMappedPort(TRINO_PORT), nullToEmpty(catalog));
+    }
+
+    @Override
+    public String getUsername() {
+        return username;
+    }
+
+    @Override
+    public String getPassword() {
+        return "";
+    }
+
+    @Override
+    public String getDatabaseName() {
+        return catalog;
+    }
+
+    @Override
+    public String getTestQueryString() {
+        return "SELECT count(*) FROM tpch.tiny.nation";
+    }
+
+    @Override
+    public SELF withUsername(final String username) {
+        this.username = username;
+        return self();
+    }
+
+    /**
+     * @deprecated This operation is not supported.
+     */
+    @Override
+    @Deprecated
+    public SELF withPassword(final String password) {
```

Comment:
if it is not supported, let's throw an error

Reply:
AFAIR it was required by JDBCDriverTest test.

Reply:
Shouldn't be the case, as we have `CockroachContainer` that throws from these methods:

https://github.com/testcontainers/testcontainers-java/blob/140fd94b361786e777ae87a8a9643b954ed68e00/modules/cockroachdb/src/main/java/org/testcontainers/containers/CockroachContainer.java#L86-L99

Reply:
I removed the method and also removed `TrinoContainerProvider.newInstance(ConnectionUrl)` override, which I think was causing problems for me.